### PR TITLE
make parameter private

### PR DESCRIPTION
**Pull request by nstarman**
Removes `from astropy.cosmology.parameter import Parameter` in favor of  `from astropy.cosmology import Parameter`, which was always the preferred import.

Discussed in #14874.

### CODE REVIEW

File: astropy/cosmology/_parameter/__init__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from . import core, descriptors
+from .core import *
+from .descriptors import *
+
+__all__: list[str] = []
+__all__ += core.__all__
+__all__ += descriptors.__all__
```
Comment (eerovaher) [inline astropy/cosmology/_parameter/__init__.py:9]: Why are these lines needed?
```suggestion

__all__: list[str] = []
__all__ += core.__all__
__all__ += descriptors.__all__
```

**Comment by eerovaher** [comment]
This has merge conflicts.

File: astropy/cosmology/_parameter/__init__.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/_parameter/__init__.py]: I suppose they're not. I moved it to `cosmology.__init__`.
